apps/web/src/actions/user-profile-mod.ts

```diff
@@ -3,48 +3,186 @@
 import { authenticatedAction } from "@/lib/safe-action";
 import { z } from "zod";
 import { db } from "db";
-import { userCommonData } from "db/schema";
+import { userCommonData, userHackerData } from "db/schema";
 import { eq } from "db/drizzle";
 import { put } from "@vercel/blob";
 import { decodeBase64AsFile } from "@/lib/utils/shared/files";
 import { revalidatePath } from "next/cache";
-import { getUser } from "db/functions";
+import { getUser, getUserByTag } from "db/functions";
+import { RegistrationSettingsFormValidator } from "@/validators/shared/RegistrationSettingsForm";
 
-// TODO: Add skill updating
 export const modifyRegistrationData = authenticatedAction(
+	RegistrationSettingsFormValidator,
+	async (
+		{
+			age,
+			gender,
+			race,
+			ethnicity,
+			isEmailable,
+			university,
+			major,
+			levelOfStudy,
+			schoolID,
+			hackathonsAttended,
+			softwareBuildingExperience,
+			heardAboutEvent,
+			shirtSize,
+			dietaryRestrictions,
+			accommodationNote,
+			github,
+			linkedin,
+			personalWebsite,
+			phoneNumber,
+			countryOfResidence,
+		},
+		{ userId },
+	) => {
+		const user = await db.query.userCommonData.findFirst({
```

Comment:
Nit: I would check the code that the authenticated action is doing. I believe it is already making this request to ensure the user has a presence in our database.

Reply:
This is to ease TypeScript with thinking `user` can be undefined on line 57. Idk another way of doing it, however.